ami/flowchart/Flowchart.py

```diff
@@ -1158,11 +1203,16 @@ async def build_views(self, nodes, ctrl=False, export=False, redisplay=False):
 
         for node in nodes:
             name = node.name()
+            state = {}
 
             node.display(topics=None, terms=None, addr=None, win=None)
-            state = {}
-            if hasattr(node.widget, 'saveState'):
-                state = node.widget.saveState()
+
+            if not node.viewed and node.widget_state:
+                node.widget.restoreState(node.widget_state)
+                state = node.widget_state
```

Comment:
I'm not sure if we want to do this anymore? We should probably just call the nodeTerm functions on the widget now instead.